### PR TITLE
SONARIAC-263 Rule S6380: Authorizing anonymous access to Azure resources is security-sensitive

### PR DESCRIPTION
**Pull request by nils-werner-sonarsource**

### CODE REVIEW

File: iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/AnonymousAccessToResourceCheck.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+   */
+  private static void checkApiManagement(CheckContext ctx, BlockTree resource) {
+    PropertyUtils.get(resource, "sign_in", BlockTree.class)
+      .ifPresentOrElse(signIn -> PropertyUtils.get(signIn, "enabled", AttributeTree.class)
+        .ifPresent(enabled -> reportOnFalse(ctx, enabled, API_MANAGEMENT_DISABLED_MESSAGE)),
+        () -> reportResource(ctx, resource, API_MANAGEMENT_MISSING_MESSAGE));
+  }
+
+  private static void checkDataFactorLinkSerciceOdata(CheckContext ctx, BlockTree resource) {
+    if (PropertyUtils.isMissing(resource, "basic_auth")) {
```
Comment (pynicolas) [inline iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/AnonymousAccessToResourceCheck.java:106]: Should this be "basic_auth" or "basic_authentication"? The Terraform docs mention "basic_authentication": https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_odata

File: iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/AnonymousAccessToResourceCheck.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+   * Report if 'sign_in' block is missing or 'sign_in->enabled' is set to false
+   */
+  private static void checkApiManagement(CheckContext ctx, BlockTree resource) {
+    PropertyUtils.get(resource, "sign_in", BlockTree.class)
+      .ifPresentOrElse(signIn -> PropertyUtils.get(signIn, "enabled", AttributeTree.class)
+        .ifPresent(enabled -> reportOnFalse(ctx, enabled, API_MANAGEMENT_DISABLED_MESSAGE)),
+        () -> reportResource(ctx, resource, API_MANAGEMENT_MISSING_MESSAGE));
+  }
+
+  private static void checkDataFactorLinkSerciceOdata(CheckContext ctx, BlockTree resource) {
```
Comment (pynicolas) [inline iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/AnonymousAccessToResourceCheck.java:105]: Typo, should probably be: "checkDataFactorLinkServiceOdata"

File: iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/AnonymousAccessToResourceCheck.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    if (PropertyUtils.isMissing(resource, "basic_auth")) {
+      reportResource(ctx, resource, DATA_FACTORY_LINKED_SERVICE_ODATA_MESSAGE);
+    }
+  }
+
+  private static void checkDataFactorLinkSerciceWebAndSftp(CheckContext ctx, BlockTree resource) {
```
Comment (pynicolas) [inline iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/AnonymousAccessToResourceCheck.java:111]: Typo, should probably be "checkDataFactorLinkServiceWebAndSftp"

File: iac-extensions/terraform/src/test/resources/checks/Azure/AnonymousAccessToResourceCheck/data_factory_linked_service.tf [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  username            = local.creds.username
+  password            = local.creds.password
+}
+
+resource "azurerm_data_factory_linked_service_sftp" "compliant_missing" {
+}
+
+
+resource "other_resource" "coverage" {
+}
```
Comment (pynicolas) [inline iac-extensions/terraform/src/test/resources/checks/Azure/AnonymousAccessToResourceCheck/data_factory_linked_service.tf:35]: I'm not sure what this test case is for.
However, I think we need a case where the name of the resource is not "azurerm_data_factory_linked_service_sftp" and which has `authentication_type = "Anonymous"`

File: iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/AnonymousAccessToResourceCheck.java [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+  protected void registerChecks() {
+    register(AnonymousAccessToResourceCheck::checkResourceAuthSettings, "azurerm_app_service",
+      "azurerm_app_service_slot",
+      "azurerm_function_app",
+      "azurerm_function_app_slot",
+      "azurerm_windows_web_app",
+      "azurerm_linux_web_app");
+    register(AnonymousAccessToResourceCheck::checkApiManagementApi, "azurerm_api_management_api");
+    register(AnonymousAccessToResourceCheck::checkApiManagement, "azurerm_api_management");
+    register(AnonymousAccessToResourceCheck::checkDataFactorLinkSerciceOdata, "azurerm_data_factory_linked_service_odata");
```
[the remark this answers is not in the document]
Comment (nils-werner-sonarsource) [reply iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/AnonymousAccessToResourceCheck.java:53]: > (Note: All resource names should be prefixed with "azurerm_" when implemented. For better readability, the prefix is removed from the ticket).

**Review by pynicolas — APPROVED** [review]

**Pull request merged**